src/pipeline_migration/registry.py

```diff
@@ -61,12 +61,21 @@ def uri_with_tag(self) -> str:
         return uri
 
 
+class AnonymousBackend:
+    """Custom backend to avoid extra HTTP requests by oras to request token"""
+
+
 class Registry(OrasRegistry):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._cache = FileBasedCache()
 
+        # oras determines backend type by checking the type of self.auth.
+        # Replace self.auth with the custom backend to avoid setting any
+        # authentication data in the header.
+        self.auth = AnonymousBackend()
```

Comment:
Looking at the oras code, this seems like a rather dirty trick which could cause issues with newer oras-py versions.

In fact, I just found https://github.com/oras-project/oras-py/commit/f9268e65237041cd0026e21c66f688ae55426ddd, which 
1. will likely break with this approach
2. may have fixed the anonymous requests situation already?

Reply:
That fix works for the case. Let's wait for a new release of oras-py and close this PR.

Reply:
It might be good to file an upstream issue to ask them for a release. They don't seem to be doing releases regularly.